app/controllers/tokens_controller.rb

```diff
@@ -0,0 +1,26 @@
+class TokensController < ApplicationController
+  before_action :require_login
+
+  def new
+    map = current_user.maps.find(params[:map_id])
+    @token = map.tokens.build
```

Comment:
I wonder if we could create a `modal.js.erb` layout that could be used in this instances. There should be a way it could pull the action's corresponding html template.

Not for a requirement for this PR, but I'd like to move more into modals.

Reply:
Could we pass the token with `render locals` instead?

Reply:
I've changed it to `render locals` in ef72c6f

Reply:
I think doing something like `modal.js.erb` could make sense as we introduce more modals, but I'll leave it off from this PR for now.